src/backend/utils/misc/queryenvironment.c

```diff
@@ -215,19 +236,26 @@ get_ENR(QueryEnvironment *queryEnv, const char *name)
  * Same as get_ENR() but just search for relation oid
  */
 EphemeralNamedRelation
-get_ENR_withoid(QueryEnvironment *queryEnv, Oid id, EphemeralNameRelationType type)
+get_ENR_withoid(QueryEnvironment *queryEnv, Oid id, EphemeralNameRelationType type, bool search)
```

Comment:
why we need search param here? Oids cannot have duplicate. When do we want to avoid finding oid in parent envs?